lib/phoenix_live_view/router.ex

```diff
@@ -170,7 +175,7 @@ defmodule Phoenix.LiveView.Router do
      as: opts[:as] || as_helper,
      private: %{phoenix_live_view: {live_view, opts}},
      alias: false,
-     metadata: %{phoenix_live_view: {live_view, action}}}
+     metadata: Map.merge(%{phoenix_live_view: {live_view, action}}, metadata)}
```

Comment:
```suggestion
     metadata: Map.put(metadata, :phoenix_live_view, {live_view, action})
```

Reply:
Our own key needs to have higher priority than user supplied ones.

Reply:
I got it.